resources/js/meyer.js

```diff
@@ -834,55 +835,33 @@ angular.module('lsAngularApp')
       $scope.carouselIndex = index;
     };
 
-    function getUrlParameter(name) {
-        name = name.replace(/[\[]/, '\\[').replace(/[\]]/, '\\]');
-        var regex = new RegExp('[\\?&]' + name + '=([^&#]*)');
-        var results = regex.exec(location.search);
-        return results === null ? '' : decodeURIComponent(results[1].replace(/\+/g, ' '));
-    };
-
-    $scope.options = {};
-
-    $scope.changeOption = function(id , key , value){
-     var decodedOptions = [];
-     var queryString = location.search.substr(1).split('&');
-
-      $timeout(function(){
-        if(queryString[0].length > 0){
-
-            angular.forEach(queryString, function(arr, idx){
-               var urlParam = arr.substr(0,arr.indexOf('='))
-               var urlParamIndex = urlParam.match(/\d+/g).join();
-               var paramVal = getUrlParameter(urlParam);
-               decodedOptions.push([urlParamIndex , paramVal, value]);
-            });
-
-            angular.forEach(decodedOptions, function(val , id){
-                $scope.options[val[0]] = val[1];
-                var selectableOption = angular.element('#selectable-option-'+ val[1] );
-                angular.element('#selected-option-'+val[0] ).text(selectableOption.text());
-            });
-
-        }else{
-         $scope.options[id] = key;
-         var selectableOption = angular.element('#selectable-option-'+key );
-         angular.element('#selected-option-'+id ).text(selectableOption.text());
-        }
-      },50);
-    };
+    var loadOptionsFromUrl = function() {
+      $scope.options = {};
+      var queryString = location.search.substr(1).split('&');
 
+      if(queryString[0].length < 1) {
+        queryString = [];
+      }
 
-    $scope.updateSlug = function(key, id, value){
+      angular.forEach(queryString, function(param){
+        var param_parts = param.split('=');
+        var option = {
```

Comment:
This is missing notes on what it's parsing, e.g., _Expects a=[1234]_ (or whatever it's expecting).